modules/EED_SquareOnsiteOAuth.module.php

```diff
@@ -837,15 +840,12 @@ public static function decryptString(string $text, bool $sandbox_mode): string
      */
     private static function cleanDataArray(array $data): array
     {
-        $sensitive_data = [
-            'access_token'  => '',
-            'refresh_token' => '',
-            'nonce'         => '',
-        ];
+        $sensitive_data = ['access_token', 'refresh_token', 'nonce'];
         foreach ($data as $key => $value) {
             $data[ $key ] = is_array($value) ? EED_SquareOnsiteOAuth::cleanDataArray($value) : $value;
+            $data[ $key ] = in_array($key, $sensitive_data) ? '***' : $value;
```

Comment:
this second line is overwriting the previous one and therefore negating what it has done

what about something like:

```suggestion
            $value = is_array($value) ? EED_SquareOnsiteOAuth::cleanDataArray($value) : $value;
            $data[ $key ] = in_array($key, $sensitive_data) ? '***' : $value;
```